recipes/m4/all/conanfile.py

```diff
@@ -1,9 +1,100 @@
-from conans import ConanFile, tools, AutoToolsBuildEnvironment
-from contextlib import contextmanager
-import functools
-import os
-
-required_conan_version = ">=1.33.0"
+from functools import lru_cache
+from os import environ, path
+
+from conan import ConanFile
+from conan.errors import ConanException
+from conan.tools.build import build_jobs
+from conan.tools.env import Environment
+from conan.tools.files import get, apply_conandata_patches, copy, rmdir, mkdir, save
+from conan.tools.gnu import AutotoolsToolchain, Autotools
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc, unix_path
+from conan.tools.scm import Version
+from conans.client import subsystems
+from conans.client.build import join_arguments
+from conans.client.subsystems import subsystem_path, deduce_subsystem, MSYS2, _escape_windows_cmd
+from conans.tools import args_to_string
+
+required_conan_version = ">=1.50.0"
+
+
+# Duck typing the subsystems._windows_bash_wrapper method, so it wil use the correct path for activation of the environments in the bash subsystem
+# issue conan-io/conan#11980
+def _windows_bash_wrapper_fix(conanfile, command, env, envfiles_folder):
+    from conan.tools.env import Environment
+    from conan.tools.env.environment import environment_wrap_command
+    """ Will wrap a unix command inside a bash terminal It requires to have MSYS2, CYGWIN, or WSL"""
+
+    subsystem = conanfile.conf.get("tools.microsoft.bash:subsystem")
+    shell_path = conanfile.conf.get("tools.microsoft.bash:path")
+    if not subsystem or not shell_path:
+        raise ConanException("The config 'tools.microsoft.bash:subsystem' and "
+                             "'tools.microsoft.bash:path' are "
+                             "needed to run commands in a Windows subsystem")
+    env = env or []
+    if subsystem == MSYS2:
+        # Configure MSYS2 to inherith the PATH
+        msys2_mode_env = Environment()
+        _msystem = {"x86": "MINGW32"}.get(conanfile.settings.get_safe("arch"), "MINGW64")
+        msys2_mode_env.define("MSYSTEM", _msystem)
```

Comment:
Why prepend and underscore with the variable `_msystem`? For me this indicates either a private variable, or a variable that is not used. Neither is true in this case.
```suggestion
        msystem = {"x86": "MINGW32"}.get(conanfile.settings.get_safe("arch"), "MINGW64")
        msys2_mode_env.define("MSYSTEM", msystem)
```